src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/TargetFrameworkMoniker.cs

```diff
@@ -2,25 +2,200 @@
+        public static readonly TargetFrameworkMoniker Net47 = Net45 with { FrameworkVersion = new Version(4, 7) };
+        public static readonly TargetFrameworkMoniker Net471 = Net45 with { FrameworkVersion = new Version(4, 7, 1) };
+        public static readonly TargetFrameworkMoniker Net472 = Net45 with { FrameworkVersion = new Version(4, 7, 2) };
+        public static readonly TargetFrameworkMoniker Net48 = Net45 with { FrameworkVersion = new Version(4, 8) };
+        public static readonly TargetFrameworkMoniker Net50 = NetCoreApp30 with { Framework = Net, FrameworkVersion = new Version(5, 0) };
+        public static readonly TargetFrameworkMoniker Net50_Windows = Net50 with { Platform = Platforms.Windows };
+        public static readonly TargetFrameworkMoniker Net50_Linux = Net50 with { Platform = Platforms.Linux };
+        public static readonly TargetFrameworkMoniker Net60 = Net50 with { FrameworkVersion = new Version(6, 0) };
+        public static readonly TargetFrameworkMoniker Net60_Linux = Net60 with { Platform = Platforms.Linux };
+        public static readonly TargetFrameworkMoniker Net60_Windows = Net60 with { Platform = Platforms.Windows };
+#pragma warning restore CA1707 // Identifiers should not contain underscores
 
-        private const string NetPrefix = "net";
+#pragma warning disable CA1034 // Nested types should not be visible
+        public static class Platforms
+        {
+            public const string Windows = "windows";
+            public const string Linux = "linux";
+        }
+#pragma warning restore CA1034 // Nested types should not be visible
 
-        public override string ToString() => Name;
+        private readonly string? _platform;
+        private readonly Version? _platformVersion;
+        private readonly string _framework;
+        private readonly Version _frameworkVersion;
 
-        public bool IsFramework => !IsNetStandard && !IsNetCore;
+        public TargetFrameworkMoniker(string framework, Version frameworkVersion)
+        {
+            _framework = framework;
+            _frameworkVersion = frameworkVersion;
+        }
 
-        public bool IsNetStandard => Name.StartsWith(NetStandardNamePrefix, StringComparison.OrdinalIgnoreCase);
+        public string Framework
+        {
+            get
+            {
+                if (IsFramework || IsNet50OrAbove)
+                {
+                    return Net;
+                }
+                else if (IsNetCoreApp)
+                {
+                    return NetCoreApp;
+                }
+                else if (IsNetStandard)
+                {
+                    return NetStandard;
+                }
+                else
+                {
+                    return Framework;
+                }
+            }
 
-        public bool IsNetCore => Name.StartsWith(NetPrefix, StringComparison.OrdinalIgnoreCase) && Name.Contains(".");
+            init => _framework = value;
```

Comment:
guard clauses for valid frameworks

Reply:
What's the need to know if it's a valid framework? It's a more closed set that platforms for sure, but I'd rather be consistent in that choice.